### PR TITLE
GRPC V2 : Get Account Non Finalized Transactions

### PR DESCRIPTION
**Pull request by tds-1**
## Purpose

- Added support for GRPC V2 `GetAccountNonFinalizedTransactions`

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

_Remove if not applicable.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: concordium-sdk/src/test/java/com/concordium/sdk/ClientV2GetAccountNonFinalizedTransactionsTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package com.concordium.sdk;
```
Comment (abizjak) [inline concordium-sdk/src/test/java/com/concordium/sdk/ClientV2GetAccountNonFinalizedTransactionsTest.java:1]: What is this example testing compared to the `concordium-sdk-examples/src/main/java/com/concordium/sdk/examples/GetAccountNonFinalizedTransactions.java`?

Comment (parv0888) [reply concordium-sdk/src/test/java/com/concordium/sdk/ClientV2GetAccountNonFinalizedTransactionsTest.java:1]: * The tests mock the grpc service and therefore can be executed without a node. 
* Tests allows to provide required inputs and tests them on various constraints. For example the account address used in the examples may / may not have any non finalized transactions when the example is executed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # Changelog
+- Added support for GRPC V2 `GetAccountNonFinalizedTransactions`
```
Comment (EmilLa1) [inline CHANGELOG.md:2]:
```suggestion
- Added support for GRPC V2 `GetAccountNonFinalizedTransactions` for querying transactions from a particular account that have not yet been finalized.
```

File: concordium-sdk-examples/src/main/java/com/concordium/sdk/examples/GetAccountNonFinalizedTransactions.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.concurrent.Callable;
+
+@Command(name = "GetAccountNonFinalizedTransactions", mixinStandardHelpOptions = true)
+public class GetAccountNonFinalizedTransactions implements Callable<Integer> {
```
Comment (EmilLa1) [inline concordium-sdk-examples/src/main/java/com/concordium/sdk/examples/GetAccountNonFinalizedTransactions.java:19]: Please add one sentence describing what the example does.

File: concordium-sdk/src/test/java/com/concordium/sdk/ClientV2GetAccountNonFinalizedTransactionsTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.AdditionalAnswers.delegatesTo;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ClientV2GetAccountNonFinalizedTransactionsTest {
```
Comment (EmilLa1) [inline concordium-sdk/src/test/java/com/concordium/sdk/ClientV2GetAccountNonFinalizedTransactionsTest.java:27]: Please add one sentence of what the file intends to test.

File: concordium-sdk/src/main/java/com/concordium/sdk/ClientV2.java [hunk 5]
```diff
@@ -279,6 +279,20 @@ public TransactionStatus getBlockItemStatus(Hash transactionHash) throws BlockNo
         }
     }
 
+
+    /**
+     * Get the list of transactions hashes for transactions that claim to be from the given account,
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/ClientV2.java:284]:
```suggestion
     * Get the list of transactions hashes for transactions from the given account,
```

**Review by EmilLa1 — COMMENTED** [review]
Looks good. 
A few nitpicky comments that we need to address, otherwise it's good to go.

File: CHANGELOG.md (hunk 2 above)
Comment (parv0888) [reply CHANGELOG.md:2]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/ClientV2.java (hunk 5 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/ClientV2.java:284]: done

File: concordium-sdk/src/test/java/com/concordium/sdk/ClientV2GetAccountNonFinalizedTransactionsTest.java (hunk 4 above)
Comment (parv0888) [reply concordium-sdk/src/test/java/com/concordium/sdk/ClientV2GetAccountNonFinalizedTransactionsTest.java:27]: done

File: concordium-sdk-examples/src/main/java/com/concordium/sdk/examples/GetAccountNonFinalizedTransactions.java (hunk 3 above)
Comment (parv0888) [reply concordium-sdk-examples/src/main/java/com/concordium/sdk/examples/GetAccountNonFinalizedTransactions.java:19]: done

**Review by EmilLa1 — APPROVED** [review]
Looks good

**Pull request merged**